server/service/src/ledger.rs

```diff
@@ -1,26 +1,80 @@
 use repository::{
-    ledger::{LedgerFilter, LedgerRepository, LedgerRow, LedgerSort},
-    Pagination, StorageConnectionManager,
+    ledger::{LedgerFilter, LedgerRepository, LedgerRow, LedgerSort, LedgerSortField},
+    EqualFilter, Pagination, PaginationOption, StorageConnectionManager,
 };
 
-use crate::usize_to_u32;
+use crate::{get_default_pagination_unlimited, i64_to_u32, usize_to_u32};
 
 use super::{ListError, ListResult};
 
-// pub const MAX_LIMIT: u32 = 5000;
-// pub const MIN_LIMIT: u32 = 1;
+pub struct ItemLedger {
+    pub ledger: LedgerRow,
+    pub balance: f64,
+}
 
 pub fn get_ledger(
     connection_manager: &StorageConnectionManager,
+    pagination: Option<PaginationOption>,
     filter: Option<LedgerFilter>,
     sort: Option<LedgerSort>,
 ) -> Result<ListResult<LedgerRow>, ListError> {
+    let pagination = get_default_pagination_unlimited(pagination);
     let connection = connection_manager.connection()?;
     let repository = LedgerRepository::new(&connection);
 
-    let rows = repository.query(Pagination::all(), filter.clone(), sort)?;
+    let rows = repository.query(pagination, filter, sort)?;
```

Comment:
I'm curious about querying for everything here? I understand you need it for balance calculations, but would there be performance concerns for an item which has many adjustments?

Reply:
This is for the stock line which hopefully shouldn't have too many entries, but if it does then the pagination can be implemented easily (: